include/rdma/fi_ext.h

```diff
@@ -123,6 +124,25 @@ struct fid_mem_monitor {
 	struct fi_ops_mem_notify *import_ops;
 };
 
+/*
+ * System logging import extension:
+ * To use, open logging fid and import.
+ */
+
+struct fi_ops_log {
+	size_t size;
+	int (*enabled)(int disable, enum fi_log_level level,
+		       enum fi_log_subsys subsys);
+	int (*ready)(int disable, enum fi_log_level level,
+		     enum fi_log_subsys subsys, uint64_t *showtime);
+	void (*log)(const char *buf);
+};
+
+struct fid_logging
+{
+	struct fid	    fid;
+	struct fi_ops_log   *ops;
```

Comment:
Shouldn't be a pointer for `struct fi_ops_log`. Will fix.

Reply:
The first field should fid, which can be used for extensibility.

Reply:
fid was not part of the structure in the second patch (restructure API calls for logging), but was added in the third patch (API changes for fi_import_fid()). 
The fid is now declared in the same patch as the one that introduces struct fid_logging